src/librustc_mir/build/mod.rs

```diff
@@ -956,7 +956,7 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
         }
 
         let body = self.hir.mirror(ast_body);
-        self.into(&Place::Local(RETURN_PLACE), block, body)
+        self.into(&Place::Base(PlaceBase::Local(RETURN_PLACE)), block, body)
```

Comment:
Maybe add a `Place::RETURN_PLACE` associated constant to `Place` to make all of these `Place::Base(PlaceBase::Local(RETURN_PLACE))` shorter.